### PR TITLE
Refactor/reports

### PR DESCRIPTION
**Pull request by adamtretera**
Based on #9 
Refactor dependencies and remove dead code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
         "fela-plugin-unit": "^11.6.0",
         "fela-plugin-validator": "^11.6.0",
         "firebase": "^8.9.1",
+        "gitmoji-cli": "^4.5.0",
```
Comment (baierjak) [inline package.json:35]: This is not needed here, install it better globally

File: src/constants/entities.js [hunk 2]
```diff
@@ -1,13 +1,12 @@
 export const EntityKeys = {
     REPORTS: 'reports',
+    REPORT: 'report',
     FILTERS: 'filters',
     AUTH: 'auth',
```
Comment (baierjak) [inline src/constants/entities.js:5]: why you need Auth in entities ?

File: src/modules/application/components/AuthRoutes/AuthRoutes.jsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { Route, Switch, Redirect } from 'react-router-dom';
 import { nanoid } from 'nanoid';
 
-import HomePage from 'modules/application/components/HomePage';
+import { HomePage } from 'modules/application';
```
Comment (baierjak) [inline src/modules/application/components/AuthRoutes/AuthRoutes.jsx:5]: 👍

File: src/modules/entities/modules/reports/services/sagas/fetchReports.js [hunk 4]
```diff
@@ -17,7 +17,9 @@ function* fetchReports() {
         const results = [];
 
         snapshot.forEach(documentSnapshot => {
-            results.push(storage.ref(documentSnapshot.id).getDownloadURL());
+            results.push(storage.ref('6HyDpbhbfOqNxxvzTXb8.png').getDownloadURL());
+            // TO WORK with official firebase
```
Comment (baierjak) [inline src/modules/entities/modules/reports/services/sagas/fetchReports.js:21]: always write // TODO - when you wanna add todo comment, for better searching through the app

File: src/modules/entities/services/reducers/api.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import { combineReducers } from 'redux';
 import { ApiKeys } from '../../../../constants';
-import { apiReducer as filters } from '../../modules/filters';
 
-import { apiReducer as reports } from '../../modules/reports';
```
Comment (baierjak) [inline src/modules/entities/services/reducers/api.js:5]: When you are in the same module (here - modules/entities) its not bad when you do it this way :) both ways are cool

File: src/modules/filters/components/Filters/Filters.jsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { FiltersFetcher, FilterBar } from 'modules/filters';
+const Filters = () => (
```
Comment (baierjak) [inline src/modules/filters/components/Filters/Filters.jsx:3]: 👍

File: src/modules/filters/components/FiltersFetcher/FiltersFetcher.jsx [hunk 7]
```diff
@@ -1,11 +1,10 @@
 import React from 'react';
 import { childrenPropType } from '@ackee/lucas';
 import useFetchFilters from '../../hooks/useFetchFilters';
-
 const FilterFetcher = ({ children }) => {
     const { success, inProgress } = useFetchFilters();
 
-    return <>{inProgress || !success ? <h1>Načítam</h1> : children}</>;
+    return <>{inProgress || !success ? null : children}</>;
```
Comment (baierjak) [inline src/modules/filters/components/FiltersFetcher/FiltersFetcher.jsx:7]: What about the new Loader? :)

File: src/modules/reports/components/ReportsFetcher/ReportsFetcher.jsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { childrenPropType } from '@ackee/lucas';
+
+import { Loader } from 'modules/ui';
+import { useReportsFetcher } from 'modules/reports';
+
+const ReportsFetcher = ({ children }) => {
+    const { success, inProgress } = useReportsFetcher();
+
+    return <Loader show={inProgress || !success}>{children}</Loader>;
```
Comment (baierjak) [inline src/modules/reports/components/ReportsFetcher/ReportsFetcher.jsx:10]: 👍

File: src/modules/ui/components/index.js [hunk 9]
```diff
@@ -2,6 +2,7 @@ export { default as Container } from './Container';
 export { default as DataCard } from './DataCard';
 export { default as CardDetail } from './CardDetail';
 export { default as NoteForm } from './NoteForm';
-export { default as FilterBar } from '../../filters/components/FilterBar';
+export { default as Loader } from './Loader';
 
+export { default as FilterBar } from '../../filters/components/FilterBar';
```
Comment (baierjak) [inline src/modules/ui/components/index.js:7]: this is some of old mistakes - its imported from different module into the UI module

File: src/services/sagas/index.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
-import { all } from "redux-saga/effects";
-import { saga as reports } from "modules/entities";
+import { all } from 'redux-saga/effects';
+import { saga } from 'modules/entities';
```
Comment (baierjak) [inline src/services/sagas/index.js:2]: 💯

File: src/constants/entities.js (hunk 2 above)
Comment (adamtretera) [reply src/constants/entities.js:5]: Deleted

File: package.json (hunk 1 above)
Comment (adamtretera) [reply package.json:35]: Already did in upcoming branch.

File: src/modules/entities/modules/reports/services/sagas/fetchReports.js (hunk 4 above)
Comment (adamtretera) [reply src/modules/entities/modules/reports/services/sagas/fetchReports.js]: ok, renamed it

File: src/modules/filters/components/FiltersFetcher/FiltersFetcher.jsx (hunk 7 above)
Comment (adamtretera) [reply src/modules/filters/components/FiltersFetcher/FiltersFetcher.jsx:7]: It just didnt feel right to have two loaders on one page.
Should I still add it here ?

File: src/modules/ui/components/index.js (hunk 9 above)
Comment (adamtretera) [reply src/modules/ui/components/index.js]: Deleted

File: src/modules/filters/components/FiltersFetcher/FiltersFetcher.jsx (hunk 7 above)
Comment (baierjak) [reply src/modules/filters/components/FiltersFetcher/FiltersFetcher.jsx:7]: yess, the loader should be just in this area of the page
every api request could take different amount of time so different components have their own loaders :)

Comment (baierjak) [reply src/modules/filters/components/FiltersFetcher/FiltersFetcher.jsx:7]: ideally filters loader should be visible in the top area of the page where loaders are and reports loader underneath in the list area

you can do it in some new MR

**Review by baierjak — APPROVED** [review]

**Pull request merged**